bin/deploy-aws

```diff
@@ -5,13 +5,33 @@ set -e;
 # abort on undefined variable
 set -u;
 
-# get current branch from git info
-PROJECT=${CI_PROJECT_NAME};
-BRANCH=$(echo "${CI_COMMIT_REF_NAME}" | cut -d'.' -f1-2); # first two parts, if tag name
+# CI_COMMIT_REF_NAME: The branch or tag name for which project is built
+# only the following options will be forwarded .gitlab-ci
+# - develop                 [for develop branch]
+# - vX.Y.Z                  [for stable releases]
+# - vX.Y.Z-(alpha|beta|rc)? [for pre-releases]
+BRANCHORTAG=${CI_COMMIT_REF_NAME};
+MINOR=$(echo "${BRANCHORTAG}" | cut -s -d'.' -f1-2); # vX.Y
+MAJOR=$(echo "${BRANCHORTAG}" | cut -s -d'.' -f1); # vX
+PRERELEASE=$(echo ${BRANCHORTAG} | cut -s -d'-' -f2); # null or (alpha|beta|rc)?
 
 # set project base
 SOURCE_PATH="build";
-DEPLOY_PATH="vf";
+DEPLOY_BASE="vf";
+
+# set deploy paths
+if [ "${BRANCHORTAG}" = "develop" ]; then
+  # if this is the develop branch, only deploy to develop folder
+  DEPLOY_PATHS="${DEPLOY_BASE}/${BRANCHORTAG}";
+elif [ -n "${PRERELEASE}" ]; then
+  # if this is a pre-release, only deploy to pre-release folder
+  DEPLOY_PATHS="${DEPLOY_BASE}/${BRANCHORTAG}";
+else
+  # if this is a stable release (ie not alpha, beta, rc)
+  # then also create major and minor versions
+  # NB: symlinks are not possible on AWS
+  DEPLOY_PATHS="${DEPLOY_BASE}/${BRANCHORTAG} ${DEPLOY_BASE}/${MINOR} ${DEPLOY_BASE}/${MAJOR}";
```

Comment:
Here also it will have to deploy to PRERELEASE folder right? as we discussed if the code is still in development and if someone wants to use it.

Reply:
testing:
```
$ CI_COMMIT_REF_NAME=develop S3_BUCKET_DEV=s3dev bin/deploy-aws
aws s3 sync build s3://s3dev/vf/develop --exclude .git* --exclude bin/* --exclude *.html --delete --cache-control max-age=60

$ CI_COMMIT_REF_NAME=v2.3.4-alpha2 S3_BUCKET_DEV=s3dev bin/deploy-aws
aws s3 sync build s3://s3dev/vf/v2.3.4-alpha2 --exclude .git* --exclude bin/* --exclude *.html --delete --cache-control max-age=60

$ CI_COMMIT_REF_NAME=v2.3.4 S3_BUCKET_DEV=s3dev bin/deploy-aws
aws s3 sync build s3://s3dev/vf/v2.3.4 --exclude .git* --exclude bin/* --exclude *.html --delete --cache-control max-age=60
aws s3 sync build s3://s3dev/vf/v2.3 --exclude .git* --exclude bin/* --exclude *.html --delete --cache-control max-age=60
aws s3 sync build s3://s3dev/vf/v2 --exclude .git* --exclude bin/* --exclude *.html --delete --cache-control max-age=60
```